scene/3d/skeleton.cpp

```diff
@@ -220,82 +220,89 @@ void Skeleton::_update_process_order() {
 	process_order_dirty = false;
 }
 
-void Skeleton::_notification(int p_what) {
+Transform Skeleton::compute_global_pose(const Bone &p_bone, const Bone *p_bonesptr, bool p_ignore_override) const {
+	if (!p_ignore_override && p_bone.global_pose_override_amount >= (1.0 - CMP_EPSILON)) {
+		return p_bone.global_pose_override;
+	} else {
+		if (p_bone.disable_rest) {
+			if (p_bone.enabled) {
 
-	switch (p_what) {
+				Transform pose = p_bone.pose;
+				if (p_bone.custom_pose_enable) {
+					pose = p_bone.custom_pose * pose;
+				}
+				if (p_bone.parent >= 0) {
 
-		case NOTIFICATION_UPDATE_SKELETON: {
+					return p_bonesptr[p_bone.parent].pose_global * pose;
+				} else {
 
-			VisualServer *vs = VisualServer::get_singleton();
-			Bone *bonesptr = bones.ptrw();
-			int len = bones.size();
+					return pose;
+				}
+			} else {
 
-			_update_process_order();
+				if (p_bone.parent >= 0) {
 
-			const int *order = process_order.ptr();
+					return p_bonesptr[p_bone.parent].pose_global;
+				} else {
 
-			for (int i = 0; i < len; i++) {
+					return Transform();
+				}
+			}
 
-				Bone &b = bonesptr[order[i]];
+		} else {
+			if (p_bone.enabled) {
 
-				if (b.global_pose_override_amount >= 0.999) {
-					b.pose_global = b.global_pose_override;
+				Transform pose = p_bone.pose;
+				if (p_bone.custom_pose_enable) {
+					pose = p_bone.custom_pose * pose;
+				}
+				if (p_bone.parent >= 0) {
+
+					return p_bonesptr[p_bone.parent].pose_global * (p_bone.rest * pose);
 				} else {
-					if (b.disable_rest) {
-						if (b.enabled) {
 
-							Transform pose = b.pose;
-							if (b.custom_pose_enable) {
-								pose = b.custom_pose * pose;
-							}
-							if (b.parent >= 0) {
+					return p_bone.rest * pose;
+				}
+			} else {
+
+				if (p_bone.parent >= 0) {
 
-								b.pose_global = bonesptr[b.parent].pose_global * pose;
-							} else {
+					return p_bonesptr[p_bone.parent].pose_global * p_bone.rest;
+				} else {
 
-								b.pose_global = pose;
-							}
-						} else {
+					return p_bone.rest;
+				}
+			}
+		}
 
-							if (b.parent >= 0) {
+		if (!p_ignore_override && p_bone.global_pose_override_amount >= CMP_EPSILON) {
```

Comment:
If I read correctly, we can't reach this, can we? All branches above are returning.

Reply:
Fixed, good catch